===== Patch instructions generated by SAVER =====
[+] { Insert: if (true) free(*(recutl_check_integrity:errors_buf)) at 3 (line 477, column 3),
        Insert: if (true) free(*(recutl_check_integrity:errors_buf->data)) at 5 (line 477, column 7) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/recutils-1.8/utils/recutl.c
+++ b/table1/benchmarks/recutils-1.8/utils/recutl.c
@@ -474,7 +474,9 @@ recutl_check_integrity (rec_db_t db,
   size_t errors_str_size;
 
   errors_buf = rec_buf_new (&errors_str, &errors_str_size);
-  if (rec_int_check_db (db, true, external_p, errors_buf) > 0)
+	tmp = rec_int_check_db (db, true, external_p, errors_buf)
+	free(errors_buf->data);
+  if (tmp > 0)
     {
       rec_buf_close (errors_buf);
       if (!verbose_p)
@@ -489,6 +491,7 @@ recutl_check_integrity (rec_db_t db,
 
       recutl_fatal (_("use --force to skip the integrity check.\n"));
     }
+	free(errors_buf);
 }
 
 bool
